File: Web/Matrimony/Matrimony.API/Models/Profiles/ProfileModel.cs
using Matrimony.Core.Enums;
using Matrimony.Framework.Models;

namespace Matrimony.API.Models.Profiles
{
    public partial record ProfileModel : BaseNopEntityModel
    {
        public int UserId { get; set; }

        public string FirstName { get; set; } = null!;

        public string? MiddleName { get; set; }

        public string? LastName { get; set; }

        public SexTypeEnum Sex { get; set; }

        public string SexValue { get { return Sex.ToString(); } }

        public string Email { get; set; } = null!;

        public string? AlternateEmail { get; set; }

        public string? PhoneNo { get; set; }

        public string? AlternatePhoneNo { get; set; }

        public string? Langauge { get; set; }

        public string? OtherInformation { get; set; }
    }
}

File: UX/src/app/components/Profile/profile.model.ts
import { BasePagedListModel } from "../../Common/Models/BasePagedListModel";
import { BaseSearchModel } from "../../Common/Models/BaseSearchModel";

export class ProfileSearchModel extends BaseSearchModel {
    name: string | undefined = "";
}

export class ProfileModel {
    id: number | undefined = 0;
    firstName: string | undefined = "";
    middleName: string | undefined = "";
    lastName: string | undefined = "";
    sex: number | undefined = 0;
    sexValue: string | undefined = "";
    email: string | undefined = "";
    alternateEmail: string | undefined = "";
    phoneNo: string | undefined = "";
    alternatePhoneNo: string | undefined = "";
    langauge: string | undefined = "";
    otherInformation: string | undefined = "";

}
